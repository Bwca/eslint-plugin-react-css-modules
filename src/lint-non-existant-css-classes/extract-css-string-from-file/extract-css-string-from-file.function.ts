import { readFileSync } from 'fs';

import { renderSync } from 'sass';

export function extractCssStringFromFile(filePath: string): string {
  /** In case of SCSS file, parse it first. */

  if (/\.scss$/.test(filePath)) {
    const scssRenderResult = renderSync({
      file: filePath,
    });
    return scssRenderResult.css.toString();
  }
  /** In case of css file, just read the contents. */
  if (/\.css$/.test(filePath)) {
    return readFileSync(filePath, 'utf8');
  }

  throw 'Unsupported file format';
}
