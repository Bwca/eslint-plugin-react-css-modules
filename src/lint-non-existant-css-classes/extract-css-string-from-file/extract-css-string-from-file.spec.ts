import { extractCssStringFromFile } from './extract-css-string-from-file.function';

describe(`${extractCssStringFromFile.name} tests for css extractor`, () => {
  it('should correctly extract css styles from scss stylesheet', async () => {
    // Arrange
    const filePath = __dirname + '/__test__/styles.scss';

    // Act
    const css = await extractCssStringFromFile(filePath);

    // Assert
    expect(css).toMatchSnapshot();
  });

  it('should correctly extract css styles from css stylesheet', async () => {
    // Arrange
    const filePath = __dirname + '/__test__/styles.css';

    // Act
    const css = await extractCssStringFromFile(filePath);

    // Assert
    expect(css).toMatchSnapshot();
  });
});
