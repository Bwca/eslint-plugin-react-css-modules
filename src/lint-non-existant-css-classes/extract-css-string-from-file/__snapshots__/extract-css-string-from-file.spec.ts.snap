// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractCssStringFromFile tests for css extractor should correctly extract css styles from css stylesheet 1`] = `
".one .two .three .four {
  color: greenyellow;
}
"
`;

exports[`extractCssStringFromFile tests for css extractor should correctly extract css styles from scss stylesheet 1`] = `
".one .two .three .four {
  color: greenyellow;
}"
`;
